docs/sphinx_api_docs_source/build_sphinx_api_docs.py

```diff
@@ -279,15 +279,14 @@ def _parse_and_process_html_to_mdx(  # noqa: C901
             else:
                 link_sidebar_entry = self._get_sidebar_entry(html_file_path=href_path)
 
-            relative_link = self._relative_path_between_documents(
-                sidebar_entry.mdx_relpath, link_sidebar_entry.mdx_relpath
-            )
+            relative_link = link_sidebar_entry.mdx_relpath
+
             without_extension = str(relative_link).replace(".mdx", "")
             if not without_extension.endswith("_class"):
                 raise Exception(  # noqa: TRY002, TRY003
                     f"Expected class mdx file path to end with _class; this could indicate a method link that will break: {without_extension}"
                 )
-            internal_ref["href"] = str(without_extension)
+            internal_ref["href"] = "/docs/reference/api/" + str(without_extension)
```

Comment:
Does this logic affect previous versions?

Reply:
no, because this only regenerates api files for current version, the old versions are already saved in the versioned_docs folder and are not modified by this script